src/App.tsx

```diff
@@ -15,36 +16,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = 'none',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState([...goodsFromServer]);
+  const [reversed, setReversed] = useState(false);
+  const [sortMethod, setSortMethod] = useState<SortType>(SortType.None);
+
+  const sortAlphabetically = () => {
+    setSortMethod(SortType.Alphabet);
+    setReversed(false);
+    setVisibleGoods([...goodsFromServer].sort((a, b) => a.localeCompare(b)));
+  };
+
+  const sortByLength = () => {
+    setSortMethod(SortType.Length);
+    setReversed(false);
+    setVisibleGoods([...goodsFromServer].sort((a, b) => b.length - a.length));
+  };
+
+  const reverseGoods = () => {
+    setReversed(prev => !prev);
+    setVisibleGoods(prevGoods => [...prevGoods].reverse());
+  };
+
+  const resetGoods = () => {
+    setSortMethod(SortType.None);
+    setReversed(false);
+    setVisibleGoods([...goodsFromServer]);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortMethod === SortType.Alphabet ? '' : 'is-light'}`}
```

Comment:
To set classes conditionally use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library